packages/lisk-p2p/src/peer.ts

```diff
@@ -12,64 +12,195 @@
  * Removal or modification of this copyright notice is prohibited.
  *
  */
-/* tslint:disable:interface-name variable-name */
-import { PeerState } from './p2p_types';
 
-// TODO: Use to create outbound socket connection inside peer object.
-// TODO: const socketClusterClient = require('socketcluster-client');
+import {
+	P2PRequestPacket,
+	P2PMessagePacket,
+	P2PResponsePacket,
+	P2PNodeStatus,
+} from './p2p_types';
+
+import socketClusterClient from 'socketcluster-client';
 
 export interface PeerConfig {
+	readonly ipAddress: string;
+	readonly wsPort: number;
+	readonly nodeStatus?: P2PNodeStatus; // TODO DELEEETETETE
 	readonly clock?: Date;
-	readonly height: number;
-	readonly id: string;
-	/* tslint:disable:next-line: no-any */
+	readonly height?: number;
 	readonly inboundSocket?: any; // TODO: Type SCServerSocket
-	readonly ipAddress: string;
 	readonly os?: string;
 	readonly version?: string;
-	readonly wsPort: number;
+}
+
+export enum ConnectionState {
+	CONNECTING = 0,
+	CONNECTED = 1,
+	DISCONNECTED = 2,
+}
+
+export interface PeerConnectionState {
+	readonly inbound: ConnectionState;
+	readonly outbound: ConnectionState;
 }
 
 export class Peer {
-	private readonly _height: number;
 	private readonly _id: string;
-	private readonly _inboundSocket: any;
+	private readonly _peerConfig: PeerConfig;
+	private readonly _height: number;
+	private _inboundSocket: any;
+	private _outboundSocket: any;
 	private readonly _ipAddress: string;
 	private readonly _wsPort: number;
+	private _nodeStatus: P2PNodeStatus | undefined;
 
 	public constructor(peerConfig: PeerConfig) {
-		this._id = peerConfig.id;
+		this._peerConfig = peerConfig;
 		this._ipAddress = peerConfig.ipAddress;
 		this._wsPort = peerConfig.wsPort;
+		this._id = Peer.constructPeerId(this._ipAddress, this._wsPort);
 		this._inboundSocket = peerConfig.inboundSocket;
-		this._height = peerConfig.height;
+		this._height = peerConfig.height === undefined ? 0 : peerConfig.height;
```

Comment:
maybe use `(!peerConfig.height) ` that will cover more invalid values and also, I thought the default is `1`?

Reply:
Default is 1 in the current core but 0 is a better default since the chain module may not have processed the genesis block yet. I think we should try to keep it 0 but if it ends up causing protocol compatibility issues then we can switch it to 1.